tests/src/test-load-geomodel.cpp

```diff
@@ -0,0 +1,258 @@
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of ASGA nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL ASGA BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *     http://www.ring-team.org
+ *
+ *     RING Project
+ *     Ecole Nationale Superieure de Geologie - GeoRessources
+ *     2 Rue du Doyen Marcel Roubault - TSA 70605
+ *     54518 VANDOEUVRE-LES-NANCY
+ *     FRANCE
+ */
+
+#include <ringmesh/ringmesh_tests_config.h>
+
+#include <geogram/basic/line_stream.h>
+
+#include <ringmesh/geomodel/geomodel.h>
+#include <ringmesh/geomodel/geomodel_geological_entity.h>
+#include <ringmesh/geomodel/geomodel_validity.h>
+#include <ringmesh/io/io.h>
+
+/*!
+ * @file Test GeoModel building
+ * @author Arnaud Botella
+ */
+
+using namespace RINGMesh;
+
+template< index_t DIMENSION >
+void load_input_geomodel( GeoModel< DIMENSION >& geomodel, const std::string& file )
+{
+    bool loaded_model_is_valid = geomodel_load( geomodel,
+        ringmesh_test_data_path + file );
+    if( !loaded_model_is_valid )
+    {
+        throw RINGMeshException( "RINGMesh Test",
+            "Failed when loading model " + geomodel.name()
+                + ": the loaded model is not valid." );
+    }
+}
+
+template< index_t DIMENSION >
+void throw_error( const GeoModel< DIMENSION >& geomodel, const std::string& entity )
+{
+    throw RINGMeshException( "RINGMesh Test", "Failed when loading model ",
+        geomodel.name(), ": the loaded model as not the correct number of ", entity );
+}
+
+void get_line( GEO::LineInput& in )
+{
+    in.get_line();
+    in.get_fields();
+}
+
+template< index_t DIMENSION >
+void check_geomodel(
+    const GeoModel< DIMENSION >& geomodel,
+    const std::string& result );
+
+template< >
+void check_geomodel( const GeoModel< 2 >& geomodel, const std::string& result )
+{
+    std::string info { ringmesh_test_data_path + "load/" + result };
+    GEO::LineInput in( info );
+    if( !in.OK() )
+    {
+        throw RINGMeshException( "TEST", "Failed to load file: ", info );
+
+    }
+    get_line( in );
+    index_t nb_corners = in.field_as_uint( 1 );
```

Comment:
You can read all the file, store information in a map EntityType -> nb_for_success. It could avoid you code duplication